core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -478,4 +478,13 @@ public void testIncludedPartitionSummaryLimit() {
     String changedPartitions = table.currentSnapshot().summary().get(SnapshotSummary.CHANGED_PARTITION_COUNT_PROP);
     Assert.assertEquals("Should set changed partition count", "2", changedPartitions);
   }
+
+  @Test(expected = UnsupportedOperationException.class)
```

Comment:
To test exceptions, we use either `TestHelpers.assertThrows` or `Assertions`. That's important so that you can add assertions about the error message, and so that you can run assertions to validate that state did not change. In this case, the the branch name should not exist and the snapshot list should be the same size.